src/test/shell/bazel/apple/bazel_objc_test.sh

```diff
@@ -117,4 +117,79 @@ EOF
       fail "Timestamp of contents of archive file should be zero"
 }
 
+function test_strip_symbols() {
+  setup_objc_test_support
+
+  rm -rf ios
+  mkdir -p ios
+
+  cat > ios/unused.h <<EOF
+#import <Foundation/Foundation.h>
+@interface UnusedClass: NSObject
+@property (nonatomic, assign) bool someProperty;
+@end
+EOF
+
+  cat > ios/unused.m <<EOF
+#import "unused.h"
+@implementation UnusedClass
+@end
+EOF
+
+  cat >ios/main.m <<EOF
+#import <UIKit/UIKit.h>
+#import "unused.h"
+
+int main(int argc, char *argv[]) {
+  NSAutoreleasePool * pool = [[NSAutoreleasePool alloc] init];
+  int retVal = UIApplicationMain(argc, argv, nil, nil);
+  [pool release];
+  return retVal;
+}
+EOF
+
+  cat >ios/BUILD <<EOF
+apple_binary(name = 'app',
+             deps = [':main'],
+             platform_type = 'ios')
+
+objc_library(name = 'main',
+             non_arc_srcs = ['main.m'],
+             deps = [':lib'])
+
+objc_library(name = 'lib',
+             srcs = ['unused.m'],
+             hdrs = ['unused.h'])
+
+EOF
+
+
+  rm -rf tools
```

Comment:
Why do you need to remake the tools directory and the crosstool? Looking at a similar test above (test_build_app) this does not appear to be necessary.

Reply:
Updated